src/common/actions/snaps.js

```diff
@@ -14,85 +11,37 @@ export const REMOVE_SNAP_ERROR = 'REMOVE_SNAP_ERROR';
 
 export function fetchUserSnaps(owner) {
   return async (dispatch) => {
-    const url = `${BASE_URL}/api/launchpad/snaps/list`;
     const query = `owner=${encodeURIComponent(owner)}`;
 
     dispatch({
-      type: FETCH_SNAPS
-    });
-
-    try {
-      const response = await fetch(`${url}?${query}`, {
-        headers: { 'Content-Type': 'application/json' },
-        credentials: 'same-origin'
-      });
-      await checkStatus(response);
-      const result = await response.json();
-      if (result.status !== 'success') {
-        throw getError(response, result);
+      type: FETCH_SNAPS,
+      [CALL_API]: {
+        types: [ FETCH_SNAPS_SUCCESS, FETCH_SNAPS_ERROR ],
```

Comment:
The main difference between my implementation and the [Redux real world example](https://github.com/reactjs/redux/blob/master/examples/real-world/src/actions/index.js) is that my example expects two action types, not three.

The reason for this is that Redux requires actions to always have a type:
> Actions must have a type property that indicates the type of action being performed.
([Redux docs: actions](http://redux.js.org/docs/basics/Actions.html)

Some Redux tools break if actions don't have a type (redux-mock-store). My solution is to use the action dispatched by the action creator to dispatch the "request type" action, instead of it [being passed as an action type](https://github.com/reactjs/redux/blob/master/examples/real-world/src/actions/index.js#L11).

I think this makes sense, because the role of middleware should be to augment other actions, not to be used in lieu of actions.